### ISSUE TITLE
Volume menu mutes at full volume and increases when should decrease volume

### ISSUE THREAD

**Issue by mick-p1982** [libretro/RetroArch#7560]
### Actual behavior

See https://github.com/libretro/beetle-psx-libretro/issues/442.

It's hard to believe this bug made it into a release, so I started with the "core" instead. They say the menu is wrong.

Bottom line is the audio volume is mapped backward from the menu, so that higher volumes go down, and max volume mutes. This contradicts the description in the menu and the configuration file I was shown.

### Steps to reproduce the bug

Go to Audio Volume, and choose an item from the menu.

### Bisect Results

[Try to bisect and tell us when this started happening]

First time I set the volume (it's very loud by default, even though the menu says -80.) The audio plugin says xaudio.

### Version/Commit

- RetroArch: [version/commit]

1.7.5

### Environment information

Windows 10

**Comment by i30817** [comment]
overflow of byte most likely.

**Comment by hizzlekizzle** [comment]
This seems to be an issue with the 'drop-down' menus. I get proper behavior if I never go into the drop-down menu at all (i.e., just using the left/right arrows from the 'audio' menu). But once I go into the drop-down, all bets are off.

**Comment by orbea** [comment]
I'm going to close this issue in favor of https://github.com/libretro/RetroArch/issues/7528 which was made a few days earlier.

**Comment by mick-p1982** [comment]
@orbea Uh... they are definitely not "invalid:duplicate" as they are completely unrelated. I don't think I experienced #7528 for what it's worth.

**Comment by mick-p1982** [comment]
P.S. How is this not fixed yet??? It's a huge, ridiculous bug as bugs go.

**Comment by orbea** [comment]
@mick-p1982 ts not yet fixed, as I can see they are both the same issue where when the drop down menus are used with audio it results with incorrect results. I strongly suspect if the other one is fixed this will be fixed too.

**Comment by orbea** [comment]
Okay, to err on the side of caution I am going to reopen this one, but please keep an eye on the similar issue to see if this one is fixed when that one is or vica versa.

**Comment by mick-p1982** [comment]
@orbea It's good to cross-reference them. But if this problem wasn't caught when the feature was implemented, I wouldn't expect it to be caught when the other problem was looked at. The problem here seems to be that the menu is mapped backward. That's more of a semantic problem than a technical one.

**Comment by orbea** [comment]
@mick-p1982 Can you still reproduce this? The referenced issue was fixed and this one might be too.

Maybe I'm blind, but I don't seen an audio volume option to reproduce this?

**Comment by mick-p1982** [comment]
I have not updated the software. If you can understand the problem, then you should be able to tell if it's still there.

Just play a game with music, and select the menu items. If they are negative numbers (in the menu) then the music volume should decrease, and not increase. -50 should be more quiet than -20. Not louder.

If someone has not addressed this bug personally, then it should probably remain open. I don't see any sense in people who are not programmers closing issues, since they can't fix the problems.

**Comment by orbea** [comment]
> If you can understand the problem, then you should be able to tell if it's still there.

This is why I asked you to reproduce it...but okay I looked again and the only option that looks relevant is `Volume Gain`. I tried `-33` using the drop down menu which was really quiet, then I tried `12` which was loud and `0` which was normal volume. It seems to work here, but I'm not sure this is how to reproduce it since your issue report is not clear on that.

Since you are not willing to reproduce it I am going to consider this fixed, anyone please let me know if I just failed to reproduce it correctly...

**Comment by mick-p1982** [comment]
That is how the feature should work. But unless you used a menu that provide like -10 -20 -30 -50 you were using a different system. The manual input field was not the bug. It's all described here and in the linked discussion. I don't know why I have to keep commenting truthfully.

**Comment by orbea** [comment]
> But unless you used a menu that provide like -10 -20 -30 -50 you were using a different system. 

Seems right so far...

>  I don't know why I have to keep commenting truthfully.

Feel free to test it out the current master yourself to make sure I tested it right.

**Comment by mick-p1982** [comment]
Okay, but I'm (like many others) just reporting a problem, as a Good Samaritan. I hope the software improves as a result. But reporting is all I can do, as I don't maintain this project. It's not like I'm a user that desperately requires a fix :)

**Comment by twinaphex** [comment]
@mick-p1982 Please just answer him directly and provide him with relevant information that he is requesting. He is a developer and he already explained to you what he can reproduce, he cannot proceed unless you provide him with the proper feedback. Also, test with the latest nightly, that way you and him are on the same page.

Either way, further confusing developers or evading direct questions is just going to aggravate them or just make them ignore this issue altogether.

**Comment by mick-p1982** [comment]
@twinaphex I've answered this ticket out of politeness several times, always repeating the information provided in the first post. I think this is unnecessary. As far the bug goes, it's probably just a string of numbers in an array that are mapped backward. Anyone should be able to trace/fix it easily. It's trivial problem unworth discussion. The second post should have been a thank you and a fix, or else silence.

**Comment by orbea** [comment]
@twinaphex I think you likely fixed this in commit https://github.com/libretro/RetroArch/commit/0e66da393078579f656f886b1adb9075ecef19bc, I at least couldn't reproduce any issues with the volume gain drop down menu...

**Comment by mick-p1982** [comment]
I think if 0e66da3 fixed it, it would mean it implements the feature correctly, whereas the deferred implementation does not. In which case, if the old implementation is still around, it should still need to be hunted down and corrected.
